packages/webpack/src/plugins/nx-webpack-plugin/lib/apply-base-config.ts

```diff
@@ -345,7 +345,35 @@ function applyNxDependentConfig(
   const externals = [];
   if (options.target === 'node' && options.externalDependencies === 'all') {
     const modulesDir = `${options.root}/node_modules`;
-    externals.push(nodeExternals({ modulesDir }));
+
+    const graph = options.projectGraph;
+    const projectName = options.projectName;
+
+    const deps = graph?.dependencies?.[projectName] ?? [];
+
+    // Collect non-buildable TS project references so that they are bundled
+    // in the final output. This is needed for projects that are not buildable
+    // but are referenced by buildable projects. This is needed for the new TS
+    // solution setup.
+    const nonBuildableWorkspaceLibs = isUsingTsSolution
+      ? deps
+          .filter((dep) => {
+            const node = graph.nodes?.[dep.target];
+            if (!node || node.type !== 'lib') return false;
+
+            const hasBuildTarget = 'build' in (node.data?.targets ?? {});
```

Comment:
Checking for `build` would probably cover most of the cases, and we should keep it as a first/fast check, but the task could also have a different name. If there's no `build` task, can we additionally check the packages' entry points to determine whether they are pointing to the source or the outputs?

FWIW, the graph nodes can contain a `metadata.js` object where the entry points are already collected, so there wouldn't be a need to read the `package.json` files.

Reply:
Fair point I can update to also check entry points if no build target.